tests/geoh5_to_las_test.py

```diff
@@ -0,0 +1,245 @@
+            workspace,
+            collar=np.r_[0.0, 10.0, 10],
+            parent=drillhole_group,
+            name="dh1",
+        )
+        drillhole_a.add_data(
+            {
+                "my_data": {
+                    "depth": np.arange(0, 50.0),
+                    "values": np.random.randn(50),
+                },
+            }
+        )
+        write_curves(drillhole_a, tmp_path, directory=False)
+
+        file = lasio.read(Path(tmp_path / "dh1.las"), mnemonic_case="preserve")
+        assert "DEPTH" in [k.mnemonic for k in file.curves]
+        file.append_curve("my_new_data", np.random.randn(50))
+        drillhole = create_or_append_drillhole(workspace, file, drillhole_group)
+
+        # New data is appended to existing drillhole
+        assert drillhole.uid == drillhole_a.uid
+        assert drillhole.get_data("my_new_data")
+
+        file = lasio.read(Path(tmp_path / "dh1.las"), mnemonic_case="preserve")
+        file.well["X"] = 10.0
+        drillhole = create_or_append_drillhole(workspace, file, drillhole_group)
+
+        # New data should be placed in a new drillhole object with augmented name
+        assert drillhole.uid != drillhole_a.uid
+        assert drillhole.name == "dh1 (1)"
+
+        file.well["WELL"] = "dh2"
+        drillhole = create_or_append_drillhole(workspace, file, drillhole_group)
+        # Same data should be read into a new
+        assert workspace.get_entity("dh2")
+
+
+def test_add_survey(tmp_path):
+    with Workspace.create(Path(tmp_path / "test.geoh5")) as workspace:
+        drillhole_group = DrillholeGroup.create(workspace, name="dh_group")
+
+        # Create a workspace
+        drillhole_a = Drillhole.create(
+            workspace,
+            collar=np.r_[0.0, 10.0, 10],
+            parent=drillhole_group,
+            name="dh1",
+        )
+        surveys = np.c_[
+            np.linspace(0, 100, 50),
+            np.ones(50) * 45.0,
+            np.linspace(-89, -75, 50),
+        ]
+        drillhole_a.surveys = surveys
+
+        drillhole_a.add_data(
+            {
+                "my_data": {
+                    "depth": np.arange(0, 50.0),
+                    "values": np.random.randn(50),
+                },
+            }
+        )
+        drillhole_to_las(drillhole_a, tmp_path, directory=False)
+        basepath = Path(tmp_path)
+        data = lasio.read(Path(basepath / "dh1.las"))
+        survey = Path(basepath / "dh1_survey.las")
+        drillhole = las_to_drillhole(workspace, data, drillhole_group, survey=survey)
+        assert np.allclose(drillhole.surveys, surveys)
+
+        survey.unlink()
+        survey = Path(basepath / "dh1_survey.csv")
+        np.savetxt(survey, surveys, delimiter=",", header="depth, dip, azimuth")
+        drillhole = las_to_drillhole(workspace, data, drillhole_group, survey=survey)
+        assert np.allclose(drillhole.surveys, surveys)
+
+
+def test_import_las(tmp_path):  # pylint: disable=too-many-locals

```

Comment:
**suggestion:** break down this function for better readability

Reply:
done.